en/ros2/px4_ros2_msg_translation_node.md

```diff
@@ -0,0 +1,110 @@
+# PX4 ROS 2 Message Translation Node
+
+<Badge type="tip" text="main (PX4 v1.16+)" /> <Badge type="warning" text="Experimental" />
+
+The message translation node allows to run applications that are compiled with one set of message versions against a PX4 with another set of message versions, without having to change either the application or the PX4 side.
+
+Specifically for this to work, topic publication/subscription/service names contain a message version in the form of `<topic_name>_v<version>`.
+
+The translation node knows about all existing message versions, and dynamically monitors the publications, subscriptions and services, and then creates translations as needed.
+
+## Installation and First Test
+
+Create a ROS 2 workspace in which to build the message translation node and its dependencies.
+
+```bash
+mkdir -p /path/to/ros_ws/src
+```
+
+Run the following helper script, it will copy the message definitions and translation node to your ROS workspace directory.
+
+```bash
+cd /path/to/ros_ws
+/path/to/PX4-Autopilot/Tools/copy_to_ros_ws.sh .
+```
+
+Build and source the workspace.
+
+```bash
+colcon build
+source /path/to/ros_ws/install/setup.bash
+```
+
+Finally, run the translation node. You should see an output similar to:
+
+```bash
+ros2 run translation_node translation_node_bin
+[INFO] [1734525720.729530513] [translation_node]: Registered pub/sub topics and versions:
+[INFO] [1734525720.729594413] [translation_node]: Registered services and versions:
+```
```

Comment:
I like lists, in particular if you add the prefix I suggest above. So

````suggestion
1. Create a ROS 2 workspace in which to build the message translation node and its dependencies.

   ```sh
   mkdir -p /path/to/ros_ws/src
   ```

2. Run the following helper script to copy the message definitions and translation node into your ROS workspace directory.

   ```sh
   cd /path/to/ros_ws
   /path/to/PX4-Autopilot/Tools/copy_to_ros_ws.sh .
   ```

3. Build and source the workspace.

   ```sh
   colcon build
source /path/to/ros_ws/install/setup.bash
   ```
4. Finally, run the translation node:

   ```sh
   ros2 run translation_node translation_node_bin
   ```

    You should see an output similar to:

   ```sh
   [INFO] [1734525720.729530513] [translation_node]: Registered pub/sub topics and versions:
   [INFO] [1734525720.729594413] [translation_node]: Registered services and versions:
   ```
````

Reply:
Awesome thanks, I agree the list is much neater. Regarding the prefix what do you mean, are you referring to "For every workspace you create, it is recommended to install and run the node:"?

Reply:
Yes!